File: src/test/java/com/github/xumeng/java/problems/easy/problem0258/Solution0258Test.java
package com.github.xumeng.java.problems.easy.problem0258;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Solution0258Test {

    private final Solution0258 solution = new Solution0258();

    @Test
    public void test1() {
        int num = 38;
        int sum = solution.addDigits(num);
        assertEquals(2, sum);
    }

    @Test
    public void test2() {
        int num = 0;
        int sum = solution.addDigits(num);
        assertEquals(0, sum);
    }

    @Test
    public void test3() {
        int num = 2147483647;
        int sum = solution.addDigits(num);
        assertEquals(1, sum);
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0258/Solution0258Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0258

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution0258Test {

    private val solution = Solution0258()

    @Test
    fun test1() {
        val num = 38
        val sum = solution.addDigits(num)
        assertEquals(2, sum)
    }

    @Test
    fun test2() {
        val num = 0
        val sum = solution.addDigits(num)
        assertEquals(0, sum)
    }

    @Test
    fun test3() {
        val num = 2147483647
        val sum = solution.addDigits(num)
        assertEquals(1, sum)
    }

}